README.rst

```diff
@@ -45,6 +45,10 @@ Basic usage::
 
 :code:`skip_smtp`: (default :code:`False`) skip the SMTP conversation with the server, after MX checks. Will automatically be set to :code:`True` when :code:`check_mx` is :code:`False`!
 
+:code:`raise_communication_errors`: Affects the SMTP verification step. If set to :code:`True`, any connection error or SMTP error message from the server will lead to a negative verification result, otherwise it will be regarded as an ambiguous result. Defaults to :code:`False`. This option is mainly used in connection with :code:`validate_email_or_fail()`, where the exception raised can be analyzed to find out the reason for the otherwise ambiguous result.
```

Comment:
I am not sure you need this. The function returning `True/False/None` doesn't need this, and in case of the function that raises errors, you should just simply raise the ambigious exceptions. No extra parameters needed.

Reply:
The idea behind this is to keep the default behavior unchanged for verify_email_or_fail for all the users that already use that function and rely on the fact that for ambiguous cases it doesn't raise an exception.

Reply:
Ah yes, that problem. I think since we're already heading towards a major release with breaking changes, might as well restructure the code to fit the new purpose and then release it as such.